segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/AppendProcessor.java

```diff
@@ -212,6 +218,27 @@ public void setupAppend(SetupAppend setupAppend) {
                         });
     }
 
+    private void populateEventSizeForIndexSegmentAppends(UUID writer, String indexSegment) {
+        try {
+            store.getStreamSegmentInfo(indexSegment, TIMEOUT)
+                    .thenAccept(properties -> {
+                        if (properties != null) {
+                            Map<AttributeId, Long> attributes =  properties.getAttributes();
+                            this.writerStates.put(Pair.of(indexSegment, writer), new WriterState(attributes.get(Attributes.ALLOWED_INDEX_SEG_EVENT_SIZE)));
+                        } else {
+                            log.trace("could not find the segment {}", indexSegment);
```

Comment:
Maybe info? Should this return an error?

Reply:
We changed it to now handle the exception.